### PR TITLE
Alvin/eigen

### PR DESCRIPTION
**Pull request by alvinsunyixiao**

### CODE REVIEW

**Review by AlchemicRonin — COMMENTED** [review]
```suggestion
  for (int i = 0; i < 20; ++i) {
    osDelay(1000);

    print("\r\n[Iter %d]: \r\n%f %f\r\n%f %f\r\n", i + 1, mat_acc(0, 0), mat_acc(0, 1), mat_acc(1, 0),
          mat_acc(1, 1));

    mat_acc = (mat_acc * mat).eval();
  }
```
Everything works fine, but after certain iterations, the matrix would overflow. So I suggest to limit the iteration num.

File: cmake/build_helper.cmake [hunk 1]
```diff
@@ -58,7 +58,7 @@ function(irm_add_arm_executable name)
         COMMAND st-flash --reset write ${BIN_FILE} 0x8000000
         DEPENDS ${name}.elf)
 
-    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    if (NOT CMAKE_BUILD_TYPE STREQUAL "Release")
```
Comment (AzulRadio) [inline cmake/build_helper.cmake:61]: Sanity check: are you changing the default cmake behavior to "DEBUG"?

**Review by AzulRadio — DISMISSED** [review]
Didn't verify but looks good. @AlchemicRonin please verify examples/eigen/main.cc if you can.

File: cmake/build_helper.cmake (hunk 1 above)
Comment (alvinsunyixiao) [reply cmake/build_helper.cmake:61]: Nope. Adding another build type `RelWithDebInfo`, which generates debug symbols with full compiler optimizations enabled.

File: boards/CMakeLists.txt [hunk 2]
```diff
@@ -26,7 +26,7 @@ target_compile_options(board_interface INTERFACE
     -ffunction-sections -fdata-sections -fmessage-length=0)
 target_link_options(board_interface INTERFACE
     -specs=nano.specs -specs=nosys.specs -Wl,--gc-sections
-    -Wl,--wrap=malloc -Wl,--wrap=free -Wl,--wrap=calloc)
+    -Wl,--wrap=malloc -Wl,--wrap=free -Wl,--wrap=calloc -Wl,--wrap=realloc)
```
Comment (AzulRadio) [inline boards/CMakeLists.txt:29]: What's this? Are we trying to overwrite malloc(), free(), calloc() and realloc() here? But didn't you say we can't overwrite them?

**Review by AzulRadio — COMMENTED** [review]
Minor change and looks good. Still need verification though.

File: boards/CMakeLists.txt (hunk 2 above)
Comment (alvinsunyixiao) [reply boards/CMakeLists.txt:29]: We can override those things with this compiler feature. Actually, only `malloc` and `free` are implemented. The `calloc` and `realloc` without implementation is to trigger a linker error when trying to link with some library that calls either of the two.

**Review by AzulRadio — APPROVED** [review]
Verified with f103 in #32

**Pull request merged**